### PR TITLE
2579: Implement link component

### PR DESCRIPTION
**Pull request by marko-ignjatovic**
### Short description

Added new `Link` component which represents merge of `AnchorLink` and `CleanLink` 

### Proposed changes

- Created `Link` component
- Removed `AnchorLink.tsx`
- Removed `CleanLink.tsx`

### Side effects

- All link components inside app

### Resolved issues

Fixes: #2579

---

### CODE REVIEW

**Review by LeandraH — COMMENTED** [review]
That looks like a great start, thank you! But there are some more places where we could replace the current links, it seems to be places where we use HTML's anchor tags like in `SharingPopup` or `react-router-dom`s `Link` like in `HeaderActionItemLink`. I found these by searching for `href`, that might help in finding more.

File: web/src/components/base/Link.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+  const getUrl = () => {
+    if (to) {
+      return to
+    }
+    if (href) {
+      return href
+    }
+    return '/'
+  }
```
Comment (steffenkleinle) [inline web/src/components/base/Link.tsx:55]: :x: This allows for wrong usage of the component since neither `to` nor `href` are required. Personally I'd favor to unify the behavior and always just have either `to` or ` href` and remove the other prop completely.

Another option to make this typesafe (also less unified/more confusing) you could do something like:
``` ts
type ToProps = {
  href: string
  to: undefined
} | {
  href: undefined
  to: string
}
```

However, I think only having either `to` or `href`` would be the better choice since
- less logic needed
- streamlined usage
- no confusion that different properties have the same result

Personally I don't mind whether we keep `to` or `href`.

File: web/src/components/base/Link.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      // @ts-expect-error wrong types from polymorphic 'as', see https://github.com/styled-components/styled-components/issues/4112
+      <StyledLink
+        as='a'
+        href={url}
+        aria-label={ariaLabel}
+        className={className}
+        data-testid='externalLink'
+        $removeHighlight={removeHighlight}
+        $adaptiveFontSize={adaptiveFontSize}
+        {...newTabProps}>
```
Comment (steffenkleinle) [inline web/src/components/base/Link.tsx:78]: :x: I think this changes behavior, right? Atm not all external links are opened in a new tabs. Maybe we should investigate whether that makes sense but this PR should not change behavior without discussion about it.

File: web/src/components/base/Link.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      </StyledAnchor>
+    )
+  }
+
+  if (isExternalUrl(url)) {
+    return (
+      // @ts-expect-error wrong types from polymorphic 'as', see https://github.com/styled-components/styled-components/issues/4112
+      <StyledLink
+        as='a'
+        href={url}
```
Comment (steffenkleinle) [inline web/src/components/base/Link.tsx:72]: :question: why do we need this separate check/typecast? Isn't this nearly exactly the same as `StyledAnchor` (except the flex and helpers which we could also introduce here).

File: web/src/components/StyledToolbarItem.tsx [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import styled from 'styled-components'
 
 import dimensions from '../constants/dimensions'
-import CleanAnchor from './CleanAnchor'
+import Link from './base/Link'
 
-const StyledToolbarItem = styled(CleanAnchor)`
+const StyledToolbarItem = styled(Link).attrs(() => ({
+  isCleanAnchor: true,
```
Comment (steffenkleinle) [inline web/src/components/StyledToolbarItem.tsx:7]: :wrench: this prop can be removed, see explanation below.

File: web/src/components/base/Link.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+type LinkProps = {
+  to?: string
+  children: ReactNode
+  ariaLabel?: string
+  className?: string
+  newTab?: boolean
+  removeHighlight?: boolean
+  adaptiveFontSize?: boolean
+  href?: string
+  isCleanAnchor?: boolean
```
Comment (steffenkleinle) [inline web/src/components/base/Link.tsx:31]: :wrench: AFAICS this is only used once with an external url and therefore this should not be necessary as it is already handled with the check for `isExternalUrl`.

File: web/src/components/ListItem.tsx [hunk 6]
```diff
@@ -32,7 +32,9 @@ const Title = styled.div`
   font-weight: 700;
 `
 
-const FullWidthLink = styled(StyledCleanLink)`
+const FullWidthLink = styled(Link).attrs(() => ({
+  stylized: true,
```
Comment (steffenkleinle) [inline web/src/components/ListItem.tsx:36]: :question: Why is this necessary?

File: web/src/components/ContactItem.tsx [hunk 7]
```diff
@@ -11,12 +10,13 @@ const Marker = styled.img`
...
 
-const Link = styled(CleanLink)`
+const StyledLink = styled(Link).attrs(() => ({
+  adaptiveFontSize: true,
+}))`
   align-items: center;
   padding-top: 4px;
   gap: 8px;
   overflow-wrap: anywhere;
-  ${helpers.adaptiveFontSize};
```
Comment (steffenkleinle) [inline web/src/components/ContactItem.tsx:19]: :question: why did you change this?

File: web/src/components/base/Link.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+const Link = ({
+  to,
+  href,
+  children,
+  ariaLabel,
+  isCleanAnchor,
+  className,
+  newTab,
+  removeHighlight,
+  adaptiveFontSize,
```
Comment (steffenkleinle) [inline web/src/components/base/Link.tsx:43]: :question: why are these questions necessary/why can't we use regular styling for this?

**Review by steffenkleinle — CHANGES_REQUESTED** [review]
Thanks for another contribution @marko-ignjatovic :tada: It is very much appreciated :)
I left some reviews with minor changes in mind, let me know if you need anything.

**Comment by steffenkleinle** [comment]
Closing in favor of #2860.
